src/inmanta/agent/executor.py

```diff
@@ -338,7 +338,8 @@ def __init__(self, envs_dir: str) -> None:
         self._locks: NamedLock = NamedLock()
         # We know that the .inmanta venv status file is touched every minute, so `61` seconds is the lowest default we can use
         self._default_expiry_cleanup = 61
-        self._cleanup_scheduler = util.Scheduler("venv_cleanup_scheduler")
+        # self._cleanup_scheduler = util.Scheduler("venv_cleanup_scheduler")
+        asyncio.ensure_future(self.start())
```

Comment:
No object should call its own start method 

The pattern we follow in most code is that the constructor is passive (it has no side effects) and sync.
 - This allows objects to be constructed with no side effects. 
 - It allows find grained control over start order. 
 - It allows validation of start conditions over many objects prior to actual start
 (many objects even had a two phase start, where pre_start does active checks )

The start method is active and async. 

The code constructing the `VirtualEnvironmentManager` should also start it.